tests/integration/test.viewadapter.js

```diff
@@ -0,0 +1,128 @@
+/* To test these adapters, you can run
+ADAPTERS=memory CLIENT=selenium:firefox npm run test */
+'use strict';
+
+var adapters = ['local'];
+
+adapters.forEach(function (adapter) {
+  describe('test.viewadapter.js-' + adapter, function () {
+    var dbs = {};
+
+    var docs = [
+      {title : 'abc', value: 1, _id: 'doc1'},
+      {title : 'def', value: 2, _id: 'doc2'}
+    ];
+
+    var ddoc = {
+      _id: '_design/index',
+      views: {
+        index: {
+          map: function mapFun(doc) {
+            if (doc.title) {
+              emit(doc.title);
+            }
+          }.toString()
+        }
+      }
+    };
+
+    function getDBNames(localStorage) {
+      var savedDbNames = Object.keys(localStorage).filter(function (key) {
+        return key.includes(dbs.name);
+      });
+
+      // This is the name of the db where view index data is stored.
+      var viewDbName = savedDbNames.find(function (dbName) {
+        return dbName.includes('-mrview-');
+      });
+      // This is the name of the db where documents are stored.
+      var docDbName = savedDbNames.find(function (dbName) {
+        return !dbName.includes('-mrview-');
+      });
+      return { viewDbName, docDbName };
+    }
+
+    beforeEach(function () {
+      dbs.name = testUtils.adapterUrl(adapter, 'testdb');
+    });
+
+    afterEach(function (done) {
+      testUtils.cleanup([dbs.name], done);
+    });
+
+    it('Create pouch with separate view adapters', function (done) {
+      var db = new PouchDB(dbs.name, {adapter: 'idb', view_adapter: 'memory'});
+
+      db.bulkDocs(docs).then(function () {
+        db.put(ddoc).then(function () {
```

Comment:
Why put the docs and ddoc separately?

Reply:
good point!I'll combine the two together:)